Tentacle/Client.swift

```diff
@@ -390,6 +393,54 @@ public final class Client {
                     .concat(response.links["next"] == nil ? SignalProducer.empty : self.fetchMany(endpoint, page: nextPage, pageSize: pageSize))
             }
     }
+
+    internal func send<Request: RequestType>(_ request: Request, to endpoint: Endpoint, method: String) -> SignalProducer<(Response, Request.Response), Error> where Request.Response == Request.Response.DecodedType {
+        let url = URL(server, endpoint)
+
+        var urlRequest = URLRequest.create(url, credentials)
+        urlRequest.httpMethod = method
+
+        let object = request.encode().JSONObject()
+        if let payload = try? JSONSerialization.data(withJSONObject: object, options: []) {
+            urlRequest.httpBody = payload
```

Comment:
I think the `httpBody` and `httpMethod` could be set inside the `create` method. (It could take a `RequestType` and encode it.)